pkg/controller/cluster/main-controller.go

```diff
@@ -342,6 +342,13 @@ func (c *Controller) syncHandler(key string) error {
 	mi.EnsureDefaults()
 	miniov1.InitGlobals(mi)
 
+	// Validate the MinIO Instance
+	if err = mi.Validate(); err != nil {
+		mi, err = c.updateMinIOInstanceStatus(ctx, mi, err.Error(), 0)
+		klog.V(2).Infof(err.Error())
+		return fmt.Errorf(err.Error())
```

Comment:
Just return error.

Reply:
Not sure why would need fmt.Errorf again.

Reply:
done